### DISCUSSION TITLE
Enhanced trains

### DISCUSSION THREAD

**Discussion by seriousmods** [scfmod/FS22_TerraFarm#44] [Ideas]
This is a totally separate idea from Terrafarm, and just for consideration. As a modder myself (with almost no scripting skills), I don't expect you to chase every idea. But for discussion: enhanced trains.

In FS17, a french scripter named FredZaza teased a train script that greatly expanded upon the in-game train system.

Here's his video from way back when:
[Train script](https://youtu.be/CsuQjt5m6nY) 

Unfortunately, it was never released or talked about since, but I've never forgotten it.

In a dreamworld, it would contain:

> The ability to spawn locomotives and cars on a track as separate objects that collide with each other (two trains can't occupy the same space)

> The ability to have multiple locomotives in a train with each drivable, but controlled by only one

> The ability to connect and disconnect cars

> The ability to switch tracks

> The ability to store cars along sidings for grain/log storage or for use in productions

> The ability to manually connect cars and/or locomotives using manual attach

> The ability for AI driven trains to respond to signals and track switching

> The ability for train cars to be enterable for individual door/unloading control

> The ability for vehicles to connect to traincars via couplers (like a maintenance of way vehicle or a loader with a coupler welded to the bucket)

Basically it would turn train locomotives into tractors on rails and train cars into trailers. Players could use store train cars at depots, then shuffle new, empty cars in to replace full ones. Players could build trains in yards, store locomotives in maintenance sheds, etc.

I'm not even sure if this idea is fully possible, but I wanted to share.

Thanks for your time!

**Comment by Throsbi** [comment]
Just ran across this comment. I was looking to see if anyone has updated the scripts from previous FS. There was a complete script set that went with Gift of Caucasus or Gold Spike or something similar. You picked which train you wanted from a menu for pickup and delivery. Was a great system. I don't remember a train running constantly on the map. Something like that would be light years ahead of what we have in FS22. 

As a side note. I have not played FS without TerraFarm since it was made available and only came back to FS because of this mod. Great work..
